samples/matter/bridge/src/bridge_manager.cpp

```diff
@@ -177,7 +201,59 @@ CHIP_ERROR BridgeManager::HandleWrite(uint16_t endpoint, ClusterId clusterId,
 		return CHIP_ERROR_INVALID_ARGUMENT;
 	}
 
-	return device->HandleWrite(clusterId, attributeMetadata->attributeId, buffer);
+	CHIP_ERROR err = device->HandleWrite(clusterId, attributeMetadata->attributeId, buffer);
+
+	/* After updating Matter BridgedDevice state, forward request to the non-Matter device. */
+	if (err == CHIP_NO_ERROR) {
+		return sBridgeManager.mDevicesMap[device]->UpdateState(clusterId, attributeMetadata->attributeId,
+								       buffer);
+	}
+	return err;
+}
+
+void BridgeManager::ReportAttributeChange(intptr_t context)
+{
+	auto path = reinterpret_cast<ConcreteAttributePath *>(context);
+
+	if (!path) {
+		return;
+	}
+
+	MatterReportingAttributeChangeCallback(*path);
+	Platform::Delete(path);
+}
+
+void BridgeManager::ScheduleReport(EndpointId endpointId, ClusterId clusterId, AttributeId attributeId)
+{
+	auto *path = Platform::New<ConcreteAttributePath>(endpointId, clusterId, attributeId);
+
+	if (!path) {
+		return;
+	}
+
+	DeviceLayer::PlatformMgr().ScheduleWork(ReportAttributeChange, reinterpret_cast<intptr_t>(path));
+}
+
+void BridgeManager::HandleUpdate(BridgedDeviceDataProvider *dataProvider, chip::ClusterId clusterId,
+				 chip::AttributeId attributeId, void *data, size_t dataSize)
+{
+	if (!dataProvider || !data) {
+		return;
+	}
+
+	std::map<BridgedDevice *, BridgedDeviceDataProvider *>::iterator it = sBridgeManager.mDevicesMap.begin();
+
+	/* The state update was triggered by non-Matter device, find related Matter Bridged Device to update it as well.
+	 */
+	while (it != sBridgeManager.mDevicesMap.end()) {
```

Comment:
nit:
```suggestion
	for (auto& item : sBridgeManager.mDevicesMap) {
```
would be shorter